src/goto-programs/validate_goto_model.h

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************\
+
+Module: Validate Goto Programs
+
+Author: Diffblue
+
+Date: Oct 2018
+
+\*******************************************************************/
+#ifndef CPROVER_GOTO_PROGRAMS_VALIDATE_GOTO_MODEL_H
+#define CPROVER_GOTO_PROGRAMS_VALIDATE_GOTO_MODEL_H
+
+#include <util/validate.h>
+
+struct goto_model_validation_optionst
+{
+  bool entry_point_exists;
+  bool function_pointer_calls_removed;
+  bool check_returns_removed;
+  bool check_called_functions;
+  bool check_last_instruction;
+  bool check_sourcecode_location;
+
+  goto_model_validation_optionst()
+    : entry_point_exists{true},
+      function_pointer_calls_removed{false},
+      check_returns_removed{false},
+      check_called_functions{false},
+      check_last_instruction{false},
+      check_sourcecode_location{false}
+  {
+  }
+
+  void disable_all_checks()
```

Comment:
It looks like `disable_all_checks()` or `enable_all_checks()` are only used for unit tests? Even if they are, I wonder if a better pattern to follow would be to either statically declare two `goto_model_validation_optionst`s, called something like `check_all_goto_model_validation_options` and `check_no_goto_model_validation_options` (of the latter is really that useful. Alternatively, rather than providing member functions or statically declared versions - just provide a second constructor, say `goto_model_validation_optionst(bool set_all)` - which would just set all the options to the value of that bool.

Reply:
Rationale was to be explicit at call location
goto_model_validation_optionst validation_options;
validation_options.disable_all_checks();
vs
goto_model_validation_optionst{false};

About the options struct - I think I would have preferred some iterable structure inside the options, but since that could be sliced 1000 ways - thought to start simple as you had suggested.